src/app/sessions/mappers/create-session-form-session-create.ts

```diff
@@ -0,0 +1,30 @@
+import * as moment from 'moment';
+import { SessionCreate } from '../models/session-create.model';
+import { CreateSessionForm } from '../models/create-session-form.model';
+
+export const SessionCreateToCreateSessionForm = (sessionCreate: SessionCreate): CreateSessionForm => {
+    const {personId, roomId, sessionTypeCode, start} = sessionCreate
+    const durationInMinutes = Math.floor(sessionCreate.duration / 60)
+    const startTime = moment(sessionCreate.start).format('HH:mm');
+    return {personId, roomId, sessionTypeCode, durationInMinutes, startDate: start, startTime}
+}
+
+export const CreateSessionFormToSessionCreate = (createSessionForm: CreateSessionForm): SessionCreate => {
+    const {personId, roomId, sessionTypeCode} = createSessionForm
+    const durationInSeconds = Math.floor(createSessionForm.durationInMinutes.valueOf() * 60)
+    const startDateWithTime = createSessionForm.startDate
+    const timeArr = createSessionForm.startTime.split(':');
+    startDateWithTime.set('hours', +timeArr[0]);
+    startDateWithTime.set('minutes', +timeArr[1]);
+
+    return {
```

Comment:
you could also do it with three dots approach (but it's not mandatory):
```
return {
        ...createSessionForm,
        caseType: null,
        sessionTypeCode,
        duration: durationInSeconds,
        start: startDateWithTime,
        id: undefined,
        userTransactionId: undefined
  }
```

Reply:
not exactly, *createSessionForm* contains property (durationInMinutes) what shouldn't be passed further

Reply:
ok, haven't noticed that